### PR TITLE
Add support for numeric time zone offsets in timestamp processor

### PR DESCRIPTION
**Pull request by jsoriano**
The ingest node `date` processor supports numeric time zone offsets,
and the `add_locale` Beat processor adds the numeric timezone in numeric
formats. Add support for these formats to the timestamp processor for
consistency with the other processors.

This is needed in #13893, where a javascript pipeline needs the timezone
set by the `add_locale` processor to generate the timestamp using the
`timestamp` processor.

### CODE REVIEW

File: libbeat/processors/timestamp/timestamp.go [hunk 1]
```diff
@@ -84,6 +84,20 @@ func newFromConfig(c config) (*processor, error) {
 	return p, nil
 }
 
+func loadLocation(timezone string) (*time.Location, error) {
+	timezoneFormats := []string{"-07", "-0700", "-07:00"}
```
Comment (kaiyan-sheng) [inline libbeat/processors/timestamp/timestamp.go]: Should timezoneFormats here be a global variable? (No need to change it if you prefer this.)

Comment (kaiyan-sheng) [reply libbeat/processors/timestamp/timestamp.go]: Will we support `Z0700   Z07:00` as well? I was reading https://yourbasic.org/golang/format-parse-string-time-date-example/ to see what are the offset/timezone format there are.

File: libbeat/processors/timestamp/timestamp_test.go [hunk 2]
```diff
@@ -199,3 +199,87 @@ func TestBuiltInTest(t *testing.T) {
...
+		},
+		"negative hour and minute offset": {
+			Timezone: "-03:30",
+			Expected: expected.Add(3*time.Hour + 30*time.Minute),
+		},
+		"negative minute offset": {
+			Timezone: "-00:30",
+			Expected: expected.Add(30 * time.Minute),
+		},
+		"negative abbreviated hour offset": {
```
Comment (kaiyan-sheng) [inline libbeat/processors/timestamp/timestamp_test.go]: maybe a test here with a given timezone format that's not supported? Like `Z0700` or `Z07:00`.

File: libbeat/processors/timestamp/timestamp.go (hunk 1 above)
Comment (jsoriano) [reply libbeat/processors/timestamp/timestamp.go]: I saw this format in the Go documentation, I think this is only useful in complete timestamps to format times with numeric zero timezones (UTC) as `Z`, or to parse `Z` as zero timestamps.

In our case here we are not parsing complete timestamps, but only timezones, that come from configuration. We could use these formats to support `Z` as a valid value for the timezone setting (as `timezone: 'Z'`), but I am not sure if we want to do that, specially when there are more usual alternatives that will work, as `timezone: 'Etc/UTC'`, or `timezone: '+0000'`.

**Comment by jsoriano** [comment]
@kaiyan-sheng `timezoneFormats` is now a global variable and I have added some error cases to the test.

**Comment by jsoriano** [comment]
jenkins, test this again please

**Pull request merged**